### PR TITLE
Refactor(Connectors): [BOA] enhance response objects

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR de-serialize the connector response related to payment checks and authentication data. 

## How did you test it?

Check if payment checks and authentication data that is sent by the connector is de serialized and displayed under connector response for confirm call in the dashboard 

1. Create a payment with Card
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: ""' \
--data-raw '{
    "amount": 100,
    "currency": "USD",
    "confirm": true,
    "customer_id": "cus_PAxm0MeCGM5TowgDKPc9",
    "payment_type": "setup_mandate",
    "email": "example@juspay.in",
      "return_url": "https://google.com",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "card_number": "4622943127013705",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "123"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrisoff Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "Chaser",
            "last_name": "Dough"
        },
        "email": "example@juspay.in"
    }
}'
```
Response
```
{
    "payment_id": "pay_XnFCntGR7e7LiHR23Pi8",
    "merchant_id": "merchant_1713863982",
    "status": "succeeded",
    "amount": 0,
    "net_amount": 0,
    "amount_capturable": 0,
    "amount_received": null,
    "connector": "bankofamerica",
    "client_secret": "pay_XnFCntGR7e7LiHR23Pi8_secret_KbcR0sWbV6y4jKFurvtr",
    "created": "2024-04-23T09:22:44.087Z",
    "currency": "USD",
    "customer_id": "cus_PAxm0MeCGM5TowgDKPc9",
    "customer": {
        "id": "cus_PAxm0MeCGM5TowgDKPc9",
        "name": null,
        "email": "example@juspay.in",
        "phone": null,
        "phone_country_code": null
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": "man_jgclWurnjMhu97Ymur3Q",
    "mandate_data": {
        "update_mandate_id": null,
        "customer_acceptance": {
            "acceptance_type": "offline",
            "accepted_at": "1963-05-03T04:07:52.723Z",
            "online": {
                "ip_address": "127.0.0.1",
                "user_agent": "amet irure esse"
            }
        },
        "mandate_type": {
            "single_use": {
                "amount": 7000,
                "currency": "USD",
                "start_date": null,
                "end_date": null,
                "metadata": null
            }
        }
    },
    "setup_future_usage": "off_session",
    "off_session": null,
    "capture_on": null,
    "capture_method": null,
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "last4": "3705",
            "card_type": null,
            "card_network": null,
            "card_issuer": null,
            "card_issuing_country": null,
            "card_isin": "462294",
            "card_extended_bin": "46229431",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "payment_checks": {
                "avs_response": {
                    "code": "Y",
                    "codeRaw": "Y"
                },
                "card_verification": {
                    "resultCode": "M",
                    "resultCodeRaw": "M"
                }
            },
            "authentication_data": null
        },
        "billing": null
    },
    "payment_token": "token_G78MMRriJMPq0lXOUrBx",
    "shipping": null,
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrisoff Street",
            "zip": "94122",
            "state": "California",
            "first_name": "Chaser",
            "last_name": "Dough"
        },
        "phone": null,
        "email": "example@juspay.in"
    },
    "order_details": null,
    "email": "example@juspay.in",
    "name": null,
    "phone": null,
    "return_url": "https://google.com/",
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": null,
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "cus_PAxm0MeCGM5TowgDKPc9",
        "created_at": 1713864164,
        "expires": 1713867764,
        "secret": "epk_606ae256d11c4e7e83d2366d0a4f65d5"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "7138641654766984903955",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": "pay_XnFCntGR7e7LiHR23Pi8_1",
    "payment_link": null,
    "profile_id": "pro_vO86u9WTWXf4ha80fMR7",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_OwnHCIqzsUmGkXUp77dL",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-04-23T09:37:44.087Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": "pm_l8DcA237GPlfKzGpC0Ki",
    "payment_method_status": null,
    "updated": "2024-04-23T09:22:46.782Z"
}
```
In dashboard, connector Api response 
```
{
  "ClientReferenceInformation": {
    "BankOfAmericaClientReferenceResponse": {
      "id": "7146552580306136503955",
      "status": "Authorized",
      "client_reference_information": {
        "code": "pay_aQvBJGxVP0sAzR3Ygjfa_1"
      },
      "processor_information": {
        "avs": {
          "code": "Y",
          "code_raw": "Y"
        },
        "card_verification": {
          "result_code": "M",
          "result_code_raw": "M"
        },
        "processor": null,
        "network_transaction_id": "*** alloc::string::String ***",
        "approval_code": "831000",
        "merchant_advice": {
          "code": "01",
          "code_raw": "M001"
        },
        "response_code": "00",
        "ach_verification": null,
        "system_trace_audit_number": "170762",
        "event_status": null,
        "retrieval_reference_number": "412313170762",
        "consumer_authentication_response": {
          "code": "2",
          "code_raw": "2"
        },
        "response_details": "ABC",
        "transaction_id": "*** alloc::string::String ***"
      },
      "processing_information": null,
      "payment_information": {
        "tokenized_card": {
          "suffix": null,
          "prefix": null,
          "expiration_month": null,
          "expiration_year": null,
          "card_type": "001"
        },
        "customer": null,
        "card": {
          "suffix": null,
          "prefix": null,
          "expiration_month": null,
          "expiration_year": null,
          "card_type": "001"
        },
        "scheme": "VISA DEBIT",
        "bin": "462294",
        "account_type": "Visa Classic",
        "issuer": "INTL HDQTRS-CENTER OWNED",
        "bin_country": "US"
      },
      "payment_insights_information": null,
      "risk_information": {
        "rules": null,
        "profile": {
          "early_decision": "ACCEPT",
          "name": "Standard mid-market profile",
          "decision": null
        },
        "score": {
          "factor_codes": ["H", "P"],
          "result": "7",
          "model_used": "default"
        },
        "info_codes": {
          "address": ["UNV-ADDR"],
          "identity_change": ["ID-M-HPOS", "ID-X-NEG", "MORPH-C"]
        }
      },
      "token_information": {
        "payment_instrument": {
          "id": "*** alloc::string::String ***"
        }
      },
      "error_information": null,
      "issuer_information": null,
      "sender_information": null,
      "payment_account_information": {
        "card": {
          "card_type": "001",
          "hashed_number": null
        },
        "features": null
      },
      "reconciliation_id": "7146552580306136503955",
      "consumer_authentication_information": {
        "eci_raw": null,
        "eci": null,
        "acs_transaction_id": null,
        "cavv": null
      }
    }
  }
}
```

2. Create a payment with Googlepay
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key:' \
--data-raw '{
    "amount": 1800,
    "currency": "USD",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "customer_id": "CustomerX",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+1",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://google.com",
    "payment_method": "wallet",
    "payment_method_type": "google_pay",
    "payment_method_data": {
        "wallet": {
            "google_pay": {
                "type": "CARD",
                "description": "Visa •••• 1111",
                "info": {
                    "card_network": "VISA",
                    "card_details": "1111"
                },
                "tokenization_data": {}
            }
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "Dubai",
            "zip": "94122",
            "country": "AE",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+97"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "AE",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "metadata": {
        "count_tickets": 1,
        "transaction_number": "5590043"
    }
}'
```
Response
```
{
    "payment_id": "pay_1yxeaMnsZLc00oM9yiLu",
    "merchant_id": "merchant_1714479295",
    "status": "succeeded",
    "amount": 1800,
    "net_amount": 1800,
    "amount_capturable": 0,
    "amount_received": 1800,
    "connector": "bankofamerica",
    "client_secret": "pay_1yxeaMnsZLc00oM9yiLu_secret_d7rDYl7TODeYiH7BDjzW",
    "created": "2024-04-30T14:12:29.597Z",
    "currency": "USD",
    "customer_id": "CustomerX",
    "customer": {
        "id": "CustomerX",
        "name": "John Doe",
        "email": "guest@example.com",
        "phone": "999999999",
        "phone_country_code": "+1"
    },
    "description": "Its my first payment request",
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "wallet",
    "payment_method_data": {
        "wallet": {},
        "billing": null
    },
    "payment_token": null,
    "shipping": {
        "address": {
            "city": "San Fransico",
            "country": "AE",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        },
        "email": null
    },
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "AE",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "Dubai",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+97"
        },
        "email": null
    },
    "order_details": null,
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "return_url": "https://google.com/",
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "google_pay",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "CustomerX",
        "created_at": 1714486349,
        "expires": 1714489949,
        "secret": "epk_c0753cfa1cef4b75bcdbf5895f806e81"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "7144863499696322703955",
    "frm_message": null,
    "metadata": {
        "count_tickets": 1,
        "transaction_number": "5590043"
    },
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": "pay_1yxeaMnsZLc00oM9yiLu_1",
    "payment_link": null,
    "profile_id": "pro_R3ZcE2GwNQFSdLb2EWla",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_B2U20vw3gnMmoZ0Z6VWb",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-04-30T14:27:29.597Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-04-30T14:12:31.515Z"
}
```
In dashboard - connector response for confirm call
```
{
  "ClientReferenceInformation": {
    "BankOfAmericaClientReferenceResponse": {
      "id": "7146557606376246604951",
      "status": "Authorized",
      "client_reference_information": {
        "code": "pay_HANScLlOyKAHexJCesmx_1"
      },
      "processor_information": {
        "avs": {
          "code": "Y",
          "code_raw": "Y"
        },
        "card_verification": null,
        "processor": null,
        "network_transaction_id": "*** alloc::string::String ***",
        "approval_code": "831000",
        "merchant_advice": {
          "code": "01",
          "code_raw": "M001"
        },
        "response_code": "00",
        "ach_verification": null,
        "system_trace_audit_number": "170794",
        "event_status": null,
        "retrieval_reference_number": "412313170794",
        "consumer_authentication_response": {
          "code": "2",
          "code_raw": "2"
        },
        "response_details": "ABC",
        "transaction_id": "*** alloc::string::String ***"
      },
      "processing_information": {
        "payment_solution": "012",
        "commerce_indicator": null,
        "commerce_indicator_label": null,
        "authorization_options": null,
        "ecommerce_indicator": null
      },
      "payment_information": {
        "tokenized_card": {
          "suffix": null,
          "prefix": null,
          "expiration_month": null,
          "expiration_year": null,
          "card_type": "001"
        },
        "customer": null,
        "card": {
          "suffix": "1111",
          "prefix": "411111",
          "expiration_month": "*** alloc::string::String ***",
          "expiration_year": "*** alloc::string::String ***",
          "card_type": "001"
        },
        "scheme": "VISA DEBIT",
        "bin": "411111",
        "account_type": "Visa Classic",
        "issuer": "CONOTOXIA SP. Z O.O",
        "bin_country": "PL"
      },
      "payment_insights_information": null,
      "risk_information": {
        "rules": null,
        "profile": {
          "early_decision": "ACCEPT",
          "name": "Standard mid-market profile",
          "decision": null
        },
        "score": {
          "factor_codes": ["B"],
          "result": "39",
          "model_used": "default_cemea"
        },
        "info_codes": {
          "address": ["MM-BIN", "UNV-ADDR"],
          "identity_change": ["ID-M-HPOS", "ID-X-HPOS"]
        }
      },
      "token_information": null,
      "error_information": null,
      "issuer_information": null,
      "sender_information": null,
      "payment_account_information": {
        "card": {
          "card_type": "001",
          "hashed_number": null
        },
        "features": null
      },
      "reconciliation_id": "7146557606376246604951",
      "consumer_authentication_information": {
        "eci_raw": null,
        "eci": null,
        "acs_transaction_id": null,
        "cavv": null
      }
    }
  }
}

```
3. Create a payment with Applepay
```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key:' \
--data-raw '{
    "amount": 1900,
    "currency": "USD",
    "confirm": true,
    
    "customer_id": "custhype1232",
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "authentication_type": "three_ds",
    "return_url": "https://google.com",
    "email": "something@gmail.com",
    "name": "Joseph Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "statement_descriptor_name": "Juspay",
    "statement_descriptor_suffix": "Router",
    "payment_method": "wallet",
    "payment_method_type": "apple_pay",
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "payment_method_data": {
      "wallet": {
         "apple_pay": {
            "payment_data": "",
            "payment_method": {
               "display_name": "",
               "network": "",
               "type": ""
            },
            "transaction_identifier": ""
         }
      }
   }
}
'
```
Response
```
{
    "payment_id": "pay_DiOCyCD80jbihcPCkNHc",
    "merchant_id": "merchant_1714479295",
    "status": "succeeded",
    "amount": 1900,
    "net_amount": 1900,
    "amount_capturable": 0,
    "amount_received": 1900,
    "connector": "bankofamerica",
    "client_secret": "pay_DiOCyCD80jbihcPCkNHc_secret_4X7JlJ2XIZgqwqc421Uj",
    "created": "2024-04-30T14:15:38.971Z",
    "currency": "USD",
    "customer_id": "custhype1232",
    "customer": {
        "id": "custhype1232",
        "name": "Joseph Doe",
        "email": "something@gmail.com",
        "phone": "999999999",
        "phone_country_code": "+65"
    },
    "description": "Its my first payment request",
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": null,
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "wallet",
    "payment_method_data": {
        "wallet": {},
        "billing": null
    },
    "payment_token": null,
    "shipping": null,
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        },
        "email": null
    },
    "order_details": null,
    "email": "something@gmail.com",
    "name": "Joseph Doe",
    "phone": "999999999",
    "return_url": "https://google.com/",
    "authentication_type": "three_ds",
    "statement_descriptor_name": "Juspay",
    "statement_descriptor_suffix": "Router",
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "payment_experience": null,
    "payment_method_type": "apple_pay",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "ephemeral_key": {
        "customer_id": "custhype1232",
        "created_at": 1714486538,
        "expires": 1714490138,
        "secret": "epk_50ee3eb74d8a42cba4f71f5e7e0a388e"
    },
    "manual_retry_allowed": false,
    "connector_transaction_id": "7144865402716199403954",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "feature_metadata": null,
    "reference_id": "pay_DiOCyCD80jbihcPCkNHc_1",
    "payment_link": null,
    "profile_id": "pro_R3ZcE2GwNQFSdLb2EWla",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_B2U20vw3gnMmoZ0Z6VWb",
    "incremental_authorization_allowed": null,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2024-04-30T14:30:38.971Z",
    "fingerprint": null,
    "browser_info": null,
    "payment_method_id": null,
    "payment_method_status": null,
    "updated": "2024-04-30T14:15:42.282Z"
}
```
In dashboard - connector's response for confirm call
```
client_reference_information{
    "id": "7144799958796957903954",
    "rootId": "7144799958796957903954",
    "reconciliationId": "7144799958796957903954",
    "submitTimeUTC": "2024-04-30T12:26:36Z",
    "merchantId": "juspay_us_sandbox",
    "applicationInformation": {
        "status": "PENDING",
        "reasonCode": 100,
        "applications": [
            {
                "name": "ics_auth",
                "reasonCode": "100",
                "rCode": "1",
                "rFlag": "SOK",
                "reconciliationId": "7144799958796957903954",
                "rMessage": "Request was processed successfully.",
                "returnCode": 1010000
            },
            {
                "name": "ics_bill",
                "status": "PENDING",
                "reasonCode": "100",
                "rCode": "1",
                "rFlag": "SOK",
                "reconciliationId": "7144799958796957903954",
                "rMessage": "Request was processed successfully.",
                "returnCode": 1260000
            },
            {
                "name": "ics_decision",
                "reasonCode": "100",
                "rCode": "1",
                "rFlag": "SOK",
                "rMessage": "Service processed successfully",
                "returnCode": 1320000
            },
            {
                "name": "ics_decision_early",
                "reasonCode": "100",
                "rCode": "1",
                "returnCode": 9999999
            }
        ]
    },
    "buyerInformation": {},
    "clientReferenceInformation": {
        "code": "pay_HsnObXV1a5KCAU55K8q9_1",
        "applicationName": "REST API",
        "applicationVersion": "1.0",
        "partner": {}
    },
    "consumerAuthenticationInformation": {
        "eciRaw": "2",
        "cavv": "ABUBhI8BYByvACzmVLv4AoABFA==",
        "strongAuthentication": {}
    },
    "deviceInformation": {},
    "errorInformation": {
        "message": "Request was processed successfully."
    },
    "installmentInformation": {},
    "fraudMarkingInformation": {},
    "merchantInformation": {
        "merchantDescriptor": {
            "name": "juspay_us_sandbox"
        }
    },
    "orderInformation": {
        "billTo": {
            "firstName": "JOSEPH",
            "lastName": "DOE",
            "address1": "1467",
            "locality": "San Fransico",
            "administrativeArea": "CA",
            "postalCode": "94122",
            "email": "something@gmail.com",
            "country": "US"
        },
        "shipTo": {},
        "amountDetails": {
            "totalAmount": "19",
            "currency": "USD",
            "taxAmount": "0",
            "authorizedAmount": "19"
        },
        "shippingDetails": {},
        "invoiceDetails": {},
        "lineItems": [
            {
                "productCode": "default",
                "taxAmount": 0,
                "quantity": 1,
                "unitPrice": 19
            }
        ]
    },
    "paymentInformation": {
        "paymentType": {
            "name": "vdcvantiv",
            "type": "credit card",
            "method": "MC"
        },
        "card": {
            "suffix": "8507",
            "prefix": "520424",
            "expirationMonth": "01",
            "expirationYear": "27",
            "type": "002"
        }
    },
    "paymentInsightsInformation": {
        "responseInsights": {},
        "ruleResults": {}
    },
    "payoutInformation": {
        "pushFunds": {},
        "pullFunds": {}
    },
    "payoutOptions": {},
    "processingInformation": {
        "paymentSolution": "MasterCard",
        "commerceIndicator": "6",
        "commerceIndicatorLabel": "spa",
        "authorizationOptions": {
            "authType": "O",
            "initiator": {
                "merchantInitiatedTransaction": {}
            }
        },
        "bankTransferOptions": {},
        "japanPaymentOptions": {},
        "fundingOptions": {
            "firstRecurringPayment": false
        },
        "ecommerceIndicator": "6"
    },
    "processorInformation": {
        "processor": {
            "name": "vdcvantiv"
        },
        "networkTransactionId": "0430MCC285073",
        "approvalCode": "831000",
        "responseCode": "00",
        "avs": {
            "code": "Y",
            "codeRaw": "Y"
        },
        "achVerification": {
            "resultCodeRaw": "00"
        },
        "systemTraceAuditNumber": "169817",
        "eventStatus": "Pending",
        "retrievalReferenceNumber": "412112169817"
    },
    "pointOfSaleInformation": {},
    "riskInformation": {
        "profile": {
            "name": "Standard mid-market profile",
            "decision": "ACCEPT"
        },
        "rules": [
            {
                "name": "Fraud Score - Monitor",
                "decision": "IGNORE"
            },
            {
                "name": "Invalid Address",
                "decision": "IGNORE"
            },
            {
                "name": "Billing-Card BIN Country Mismatch",
                "decision": "IGNORE"
            }
        ],
        "score": {
      "factorCodes

": ["H"],
      "result": 0
        }
    },
    "recipientInformation": {},
    "senderInformation": {
        "paymentInformation": {
            "customer": {},
            "paymentType": {
                "name": "vdcvantiv",
                "type": "credit card",
                "method": "MC"
            },
            "card": {
                "suffix": "8507",
                "expirationMonth": "01",
                "expirationYear": "27",
                "type": "002"
            },
            "invoice": {},
            "accountFeatures": {},
            "fluidData": {
                "descriptor": "cybersource"
            },
            "issuerInformation": {},
            "paymentAccountInformation": {}
        }
    },
    "tokenInformation": {},
    "_links": {
        "self": {
            "href": "https://apitest.merchant-services.bankofamerica.com/tss/v2/transactions/7144799958796957903954",
            "method": "GET"
        }
    }
}
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 1]
```diff
@@ -177,7 +177,7 @@ pub struct CaptureOptions {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct BankOfAmericaPaymentInstrument {
-    id: Secret<String>,
+    id: Option<Secret<String>>,
```
Comment (deepanshu-iiitu) [inline crates/router/src/connector/bankofamerica/transformers.rs:180]: Why is this field made an option?

**Review by deepanshu-iiitu — CHANGES_REQUESTED** [review]

**Review by deepanshu-iiitu — DISMISSED** [review]
LGTM

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 2]
```diff
@@ -722,6 +745,44 @@ pub struct ClientReferenceInformation {
 pub struct ClientProcessorInformation {
     avs: Option<Avs>,
     card_verification: Option<CardVerification>,
+    processor: Option<ProcessorResponse>,
+    network_transaction_id: Option<String>,
```
Comment (SamraatBansal) [inline crates/router/src/connector/bankofamerica/transformers.rs:749]:
```suggestion
    network_transaction_id: Option<Secret<String>>,
```

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 3]
```diff
@@ -722,6 +745,44 @@ pub struct ClientReferenceInformation {
...
+    approval_code: Option<String>,
+    merchant_advice: Option<MerchantAdvice>,
+    response_code: Option<String>,
+    ach_verification: Option<AchVerification>,
+    system_trace_audit_number: Option<String>,
```
Comment (SamraatBansal) [inline crates/router/src/connector/bankofamerica/transformers.rs:754]: Can you please confirm none of the structs contain sensitive data, if so can you please mask it as a secret

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 4]
```diff
@@ -735,6 +796,39 @@ pub struct CardVerification {
...
+    result: Option<RiskResult>,
+    model_used: Option<String>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum RiskResult {
+    String(String),
+    Int(u64),
+}
```
Comment (SamraatBansal) [inline crates/router/src/connector/bankofamerica/transformers.rs:824]: Is this tested locally with both the cases?

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 5]
```diff
@@ -735,6 +796,39 @@ pub struct CardVerification {
...
+    factor_codes: Option<Vec<String>>,
+    result: Option<RiskResult>,
+    model_used: Option<String>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum RiskResult {
+    String(String),
+    Int(u64),
```
Comment (SamraatBansal) [inline crates/router/src/connector/bankofamerica/transformers.rs:823]:
```suggestion
    StringVariant(String),
    IntVariant(u64),
```

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Review by SamraatBansal — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Pull request merged**